core/src/main/java/org/elasticsearch/common/settings/SettingsModule.java

```diff
@@ -112,7 +112,8 @@ public SettingsModule(Settings settings, List<Setting<?>> additionalSettings, Li
                 builder.append("Please ensure all required values are updated on all indices by executing: ");
                 builder.append(System.lineSeparator());
                 builder.append(System.lineSeparator());
-                builder.append("curl -XPUT 'http://localhost:9200/_all/_settings?preserve_existing=true' -d '");
+                builder.append("curl -XPUT -H 'Content-Type: application/json' ");
+                builder.append("'http://localhost:9200/_all/_settings?preserve_existing=true' -d '");
```

Comment:
Hmm...I wonder if we should have this here at all. There are so many cases where this is not correct and I think that could be very frustrating for users:
 - node does not bind to localhost
 - node does not accept HTTP (only HTTPS)
 - node requires basic authentication
 - user does not have `curl` (Windows?)
 - node does not bind to port 9200

Reply:
this could be circumvented by just using the console syntax.. main question is though, if we want a full request in here